### PR TITLE
Road widening, part 2: geometry

### PR DESCRIPTION
**Pull request by dabreegster**
I heard the solution to all traffic problems is more lanes, so here:

screencast

The chain of effects is: when we modify a road's width, it affects the polygon of the two connected intersections. That new intersection polygon in turn causes all connected roads to have their center line trimmed differently. Luckily, that's as far as the effect spreads out.

### CODE REVIEW

File: map_gui/src/render/map.rs [hunk 1]
```diff
@@ -39,8 +39,9 @@ pub struct DrawMap {
     pub show_zorder: isize,
 
     quadtree: QuadTree<ID>,
-    // Remember these so we can delete lanes.
+    // Remember these so we can modify the bounding box of some objects.
     lane_ids: HashMap<LaneID, aabb_quadtree::ItemId>,
+    intersection_ids: HashMap<IntersectionID, aabb_quadtree::ItemId>,
```
Comment (michaelkirk) [inline map_gui/src/render/map.rs:44]: Alternatively, we could combine `lane_ids`/`intersection_ids` into a single: `HashMap<ID , aabb_quadtree::ItemId>`

... orrr I can ask again if we add another type that needs to be reverse lookup'd

File: map_gui/src/render/map.rs [hunk 2]
```diff
@@ -466,4 +471,16 @@ impl DrawMap {
         let item_id = self.lane_ids.remove(&l).unwrap();
         self.quadtree.remove(item_id).unwrap();
```
Comment (michaelkirk) [inline map_gui/src/render/map.rs:472]: Question re: `Lane`s vs `Road`s.

This is maybe more related to the previous PR, but don't we also need to update the quadtree's bbox for the lane's Road whenever we add or remove a lane?

Orrrr do we not even use the `Road` in the quadtree and only use `Lane`s?

**Review by michaelkirk — APPROVED** [review]

File: map_gui/src/render/map.rs (hunk 1 above)
Comment (dabreegster) [reply map_gui/src/render/map.rs]: Done! I get the feeling that we'll allow editing parking lots at some point. ;)

File: map_gui/src/render/map.rs (hunk 2 above)
Comment (dabreegster) [reply map_gui/src/render/map.rs:473]: Good catch -- fixed. Already making use of that `quadtree_ids` refactor. :)

To test this, I made a very wide road and used the unzoomed road mouseover. I had to make it reaaaaaally wide to make sure the previous quadtree bin was getting exceeded:
Screenshot from 2021-04-10 11-48-03
I think some graves in this cemetery may have been displaced by this change. :(

**Comment by dabreegster** [comment]
Thanks for the review! Next step may be the hardest: re-snapping all those pesky objects that need to know their sidewalk...

**Pull request merged**